Homepage is ready for code review! ***Do not merge***

JKinsman opened the pull request:
@jaypoulz , this branch addresses #1465 .  It is not ready to pull because we still do not have the proper image assets.

Change Summary:
-New Homepage
-Coverage to 89.86% (I did not implement tests for the "Report a Bug" feature, so the other tests that have been written should boost coverage back above 91%.)

jaypoulz commented on app/assets/javascripts/home.js.coffee:
```diff
@@ -3,8 +3,133 @@
+    #Resize UI components based upon Bootstrap and screen size
+    ###
+    resizeUI = () ->
+      if screenSize('md') or screenSize('sm')
```
Ewwwwww grosssssssss make it staph make it stahp yuck yuck yuck get it off. Instead of having tons of jqueries doing the style, we should probably be making use of bootstrap + scss -> look at media queries. Ideally, none of the styling has to be done dynamically with coffeescript. http://getbootstrap.com/css/#grid-example-fluid

jaypoulz commented on test/CSVs/Konqi.svg:
```diff
@@ -0,0 +1,1068 @@
+         id="path624" />
+      <path
+         style="font-size:12.000000;fill:url(#linearGradient2359);fill-rule:evenodd;stroke:#000000;stroke-width:2.5000000"
+         d="M -521.41970,213.50265 L -513.40640,205.48930 L -504.39130,197.47594 L -497.37970,195.97343 
```
1068 lines of konqi..... lol

jaypoulz commented on test/unit/visualization_test.rb:
```diff
@@ -27,4 +28,47 @@ def setup
-end
+
+  # Passes if Visualization ID is nil
+  test 'viz_id is nil' do
```
please use vis_id is nil. This is a pet peeve of mine

jaypoulz commented:
Mostly looks fine except for a few points.
1. $ outside the parens
2. Avoid styling within the JS/CS as much as is humanly possible
3. Tests look mostly fine, but I'd have to run them to make sure
4. Is the magic scroll thing really necessary? (I might change my mind after I play with it)
5. Finally, is underscore even being used?

JKinsman commented:
1. No problem!
2.  I'll look into media queries.  I'm not sure they'll do exactly what I need, but I agree the resizeUI function is, yucky, at best. 
3.  They all pass, and I promise they're doing the right thing
4.  I think you'll like it :P
5.  Not anymore!  I'll remove the underscore gem as a dependency

kcarcia commented:
@JKinsman just a heads up, you will need to pull and re-commit at some point.

JKinsman commented:
@jaypoulz @kcarcia I believe I have addressed the concerns mentioned above:
1.  $ outside the parens
2.  I took a *bunch* of styling out of the JS and replaced it with media queries.  The lines left in the resizeUI function are the ones that seemed unreasonable to do in CSS.
3.  N/A, the tests have already been merged :smile: 
4.  N/A
5.  Removed underscore.js from the Gemfile

jaypoulz commented on app/assets/stylesheets/home.css.scss:
```diff
@@ -2,3 +2,401 @@
+        height: 90px;
+        width: 90px;
+      }
+      .img-rounded{
```
can you please add a space between the tag and the open {

jaypoulz commented on app/assets/stylesheets/home.css.scss:
```diff
@@ -2,3 +2,401 @@
+      height: 45vh;
+      overflow-y: scroll;
+    }
+    /*
```
No comments in production code unless they're there to clarify

jaypoulz commented on app/assets/stylesheets/home.css.scss:
```diff
@@ -2,3 +2,401 @@
+    }
+    .ternary-content{
+    }
+    /*
```
ditto

jaypoulz commented on app/assets/stylesheets/home.css.scss:
```diff
@@ -2,3 +2,401 @@
+        clear: '';
+      }
+    }
+    /*
```
ditto

jaypoulz commented on app/views/home/index.html.erb:
```diff
@@ -1,120 +1,243 @@
+      
+      <!---First Slide-->
+      <div class="item active">
+        <img src="<%= image_path 'create.jpg'%>" class="carousel-img" alt="First slide">
```
space before the %>

jaypoulz commented on app/views/home/index.html.erb:
```diff
@@ -1,120 +1,243 @@
+      
+      <!---Second Slide-->
+      <div class="item">
+        <img src="<%= image_path 'Share.jpg'%>" class="carousel-img" alt="Second slide">
```
ditto

jaypoulz commented on test/functional/projects_controller_test.rb:
```diff
@@ -38,6 +38,8 @@ class ProjectsControllerTest < ActionController::TestCase
 
     get :show, id: @project_two, format: 'json'
     assert_response :success
+    # FIXME
```
Nononono. No commenting out the validation. Actually fix the styling on the page, don't comment out the test.

jaypoulz commented on test/functional/visualizations_controller_test.rb:
```diff
@@ -53,6 +54,8 @@ class VisualizationsControllerTest < ActionController::TestCase
     get :show, { id: @vis2.id, presentation: true },  user_id: @kate.id
     assert_response :success
+    # FIXME
+    # assert_valid_html response.body
```
Grrrrrrrrr :imp:

JKinsman commented:
@jaypoulz I'm sorry!  I must have incorrectly resolved some of the merge conflicts, and accidentally reset them to the way were in my old homepage patch before you made all the HTML fixes.  Whoops!  I'll uncomment out all the invalid HTML assertions, and fix whatever residuals errors pop up :smile:

JKinsman commented:
@jaypoulz The homepage is ready for review again :smile:

stowellm commented:
@JKinsman - the part that says:
`We hold workshops for teachers to learn how to use iSENSE in the classroom, and for students to learn about scientific data collection and analysis.`

There should be no comma before the word "and."  Commas go where the piece following "and" makes a logical sentence - `For students to learn about scientific data collection and analysis.` is not one.

stowellm commented:
I also still personally think the news articles are **huge** :sob:

jaypoulz commented:
@JKinsman mo merge conflicts, mo money

jaypoulz commented:
@bdonald25 there is no way this can be merged now - this is so stale, you'd be better off doing it from scratch

bdonald25 commented:
@jaypoulz The overall consensus of the lab was to keep the homepage we have now and improve it for mobile devices rather than move to this one. We did that a few months ago so this one is officially dead. RIP Jakes homepage.
